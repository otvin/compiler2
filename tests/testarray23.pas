program testarray23(output);
{test local variable arrays}

type myarr=array[-5..5] of boolean;

var
	a:myarr;
	i:integer;

procedure printarr(d:myarr);
var i:integer;
begin
	i := -5;
	while i <=5 do begin
		write(d[i], ' ');
		i := i + 1;
	end;
	writeln;
end;

procedure invertarr(b:myarr);
var
	i:integer; c:myarr;
begin
	i := -5;
	while i <= 5 do begin
		c[-1 * i] := b[i];
		i := i + 1;
	end;
	printarr(b);
	writeln('inverted');
	printarr(c);
end;

begin
	i := -5;
	while i <= 5 do begin
		a[i] := (i < -3) or (i = 1);
		i := i + 1;
	end;
	invertarr(a);
end.
