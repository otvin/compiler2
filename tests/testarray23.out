TRUE TRUE FALSE FALSE FALSE FALSE TRUE FALSE FALSE FALSE FALSE 
inverted
FALSE FALSE FALSE FALSE TRUE FALSE FALSE FALSE FALSE TRUE TRUE 
